python/paddle/fluid/dygraph/jit.py

```diff
@@ -616,105 +623,118 @@ def train(layer, loader, loss_fn, opt):
-        # record var info
-        extra_info_dict = dict()
-        if param_or_buffer.name in state_names_dict:
-            extra_info_dict['structured_name'] = state_names_dict[
-                param_or_buffer.name]
-        extra_info_dict['stop_gradient'] = param_or_buffer.stop_gradient
-        if isinstance(param_or_buffer, ParamBase):
-            extra_info_dict['trainable'] = param_or_buffer.trainable
-        extra_var_info[param_or_buffer.name] = extra_info_dict
-
-    # 5. save inference model
-    from paddle.fluid.io import save_inference_model
-
-    # construct new save_inference_model arguments
-    model_path = dirname
-    # NOTE(chenweihang): because prefix contains model and params filename,
-    # so we don't support set model_filename & params_filename 
-    model_filename = file_prefix + INFER_MODEL_SUFFIX
-    params_filename = file_prefix + INFER_PARAMS_SUFFIX
-
-    with scope_guard(scope):
-        save_inference_model(
-            dirname=model_path,
-            feeded_var_names=input_var_names,
-            target_vars=output_vars,
-            executor=Executor(_current_expected_place()),
-            main_program=concrete_program.main_program.clone(),
-            model_filename=model_filename,
-            params_filename=params_filename,
-            export_for_deployment=configs._export_for_deployment,
-            program_only=configs._program_only)
-
-        # NOTE(chenweihang): [ Save extra variable info ]
-        # save_inference_model will lose some important variable information, including:
-        #   - Variable name and correspondence (when saved variables as one file)
-        #   - Variable.stop_gradient information
-        #   - Which persistent variable are parameter and which are not
-        #   - Parameter.trainable information
-        #
-        # The lost information cannot be recovered when it is loaded again, 
-        # so if we want to perform fine-tune after loading, we may need to 
-        # configure redundant information to proceed.
-        #
-        # Due to compatibility issues, we cannot change the original storage structure, 
-        # but we can save these information in `jit.save` without changing the original 
-        # storage to improve user experience. So we save extra information into
-        # file `***.pdiparams.info`
-        extra_var_info_path = path + INFER_PARAMS_INFO_SUFFIX
-        with open(extra_var_info_path, 'wb') as f:
-            pickle.dump(extra_var_info, f, protocol=2)
+        # construct new save_inference_model arguments
+        model_path = dirname
+        # NOTE(chenweihang): because prefix contains model and params filename,
+        # so we don't support set model_filename & params_filename 
+        if 'forward' == attr_func:
+            model_filename = file_prefix + INFER_MODEL_SUFFIX
+            params_filename = file_prefix + INFER_PARAMS_SUFFIX
+            extra_var_info_path = path + INFER_PARAMS_INFO_SUFFIX
```

Comment:
may be we only need to save one var_info, because all methods of one Layer share same state_dict

Reply:
Done, thanks.